pkgs/development/python-modules/temporalio/default.nix

```
@@ -0,0 +1,72 @@
+{
+  lib,
+  buildPythonPackage,
+  cargo,
+  fetchFromGitHub,
+  poetry-core,
+  protobuf,
+  python-dateutil,
+  rustc,
+  rustPlatform,
+  setuptools,
+  setuptools-rust,
+  types-protobuf,
+  typing-extensions,
+}:
+
+buildPythonPackage rec {
+  pname = "temporalio";
+  version = "1.8.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "temporalio";
+    repo = "sdk-python";
+    rev = "refs/tags/${version}";
+    hash = "sha256-xJIb+U3hV6DDvfbfY20U/4rE2z0XzaESkehtcvG6m50=";
+    fetchSubmodules = true;
+  };
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit src;
+    sourceRoot = "${src.name}/${cargoRoot}";
+    hash = "sha256-+A226LfxqmUA/CMXwx8osgIRJhfKtB8KcCSBkbbXOHk=";
+  };
+
+  cargoRoot = "temporalio/bridge";
+
+  patches = [ ./protoc-wheel-removal.diff ];
+
+  build-system = [
+    poetry-core
+    setuptools
+    setuptools-rust
+  ];
+
+  dependencies = [
+    protobuf
+    python-dateutil
+    types-protobuf
+  ];
```

```suggestion
  dependencies = [
    protobuf
    types-protobuf
    typing-extensions
  ] ++ lib.optional (pythonOlder "3.11") python-dateutil;
```
https://github.com/temporalio/sdk-python/blob/b16215104069f798b79592679d8a16dd3d702883/pyproject.toml#L33-L41